src/components/app.js

```diff
@@ -19,51 +20,88 @@ import NewIssue from 'components/issues/new-issue/NewIssue';
 
 import {
   getUserInfoWithToken,
-  saveTokenToLocalStorage
+  saveTokenToLocalStorage,
+  getUserTokenWithCode
 } from 'actions/user.actions';
 
 type ConnectedProps = {
   getUserInfoWithToken: () => void,
-  saveTokenToLocalStorage: () => void
+  saveTokenToLocalStorage: () => void,
+  isAuthenticated: string | null
 };
 
 type OwnProps = {};
 
 class App extends React.Component<ConnectedProps & OwnProps> {
   componentWillMount() {
     const user = get('user', loadFromStorage('auth'));
-    if (user) {
+
+    if (user && !isNull(user.token)) {
```

Comment:
Since you only need the user token, there is no need to get the entire user object.
Instead of doing this check you can pull the token directly using the get function above and just check if you have a token